interp/type.go

```diff
@@ -835,17 +835,19 @@ func (t *itype) methods() methodSet {
 			res[m.Name] = m.Type.String()
 		}
 	case ptrT:
-		// Consider only methods where receiver is a pointer to type t
-		for _, m := range t.val.method {
-			if m.child[0].child[0].lastChild().typ.cat == ptrT {
-				res[m.ident] = m.typ.TypeOf().String()
-			}
+		for k, v := range t.val.methods() {
+			res[k] = v
 		}
-	default:
-		for _, m := range t.method {
-			res[m.ident] = m.typ.TypeOf().String()
+	case structT:
+		for _, f := range t.field {
+			for k, v := range f.typ.methods() {
+				res[k] = v
+			}
 		}
 	}
+	for _, m := range t.method {
+		res[m.ident] = m.typ.TypeOf().String()
+	}
```

Comment:
```suggestion
```
If we move these lines according to my suggestions, it seems like all the tests in ./interp still pass.
So I don't know if my suggestion is correct (because I don't fully understand the whole case), or if we are missing tests, but I think if my suggestion is correct, then we should apply it, because it makes it clear that this for loop is only needed in the struct case.

Reply:
The following test demonstrates the need to keep this block as is.

```go
package main
  
type T int

func (t T) Error() string { return "T: error" }

var invalidT T

func main() {
    var err error
    if err != invalidT {
        println("ok")
    }
}

// Output:
// ok
```

The block handles the method set for arbitrary types, in addition of specific actions for pointers, interfaces, values, and struct. Here the example defines an alias `int` which satisifies the `error` interface.

Note that it triggers a panic in the `notEqual()` operator, but it is outside the scope of this PR. By moving the block as you propose we fail in a `mismatch types` error which shouyld not happen.

P.S. Just filing a new issue for the above test.